src/Features/CSharp/Portable/ValidateFormatString/CSharpValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.ValidateFormatString;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp.ValidateFormatString
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpValidateFormatStringDiagnosticAnalyzer :
+        AbstractValidateFormatStringDiagnosticAnalyzer<SyntaxKind>
+    {
+        protected override ISyntaxFactsService GetSyntaxFactsService()
+            => CSharpSyntaxFactsService.Instance;
+
+        protected override SyntaxKind GetInvocationExpressionSyntaxKind()
+            => SyntaxKind.InvocationExpression;
+
+        protected override ITypeSymbol GetArgumentExpressionType(
+            SemanticModel semanticModel, 
+            SyntaxNode argsArgument)
+                => semanticModel.GetTypeInfo(((ArgumentSyntax)argsArgument).Expression).Type;
```

Comment:
Seems unneeded.  Why could we not do this in the base type once you've gotten the expression off of the argument?